module-2/backend/week_two.md

```diff
@@ -25,20 +37,55 @@ end
 ```
 
 Now how would you find the owner of the team with an id of 4?
+* Team.find(4).owner
+or possibly
+* Team.find(4).owner.name
+
 
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+![SCHEMA DESIGNER DIAGRAM OF RELATIONSHIPS](http://i.imgur.com/65scG0D.jpg)
+
 6. Define foreign key, primary key, and schema.
+* Foreign key lives on a table and refers back to a primary key on another table. Foreign Key lives on the "belongs_to" side of a "one-to-many" relationship. Also can be involved in a many to many relationship using "belongs to" and "belongs to through"
+* Primary Key lives on a table. It is auto assigned when a new record is created (unless table is set to operate otherwise). Primary keys can not be reused. If deleted that number is dead forever.
+* The schema is the shecmatic document that shows you, and tells your app, how the tables in your database are set up and in most cases how they should interact with eachother. Deeper relationship functionality can be expanded in the model file if desired, but the Schema is the real heart of the tables. Without migrations there would be o tables/schema.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+* This is, again, what sounds like a one to many relationship (if it's as simple as is being stated here). The fireign key references the primary key on the table it points to.
+
 8. What are the parts of an HTTP response?
+* Request to database
+* Router
+* Controller
+* Model
+* Database
+* Return to model with info
+* Populate desired view with data from model
+* Return to controller for info on redirect or render
+* Controller returns a response to the client
+* Client potentially makes more requests based on what the controller sent back on previous request.
```

Comment:
This to me is a description of the request response cycle. When we're talking just about the response I think we're looking for:

* status line
* headers
* body